### PR TITLE
upgraded register transformer to babel transform-system-register

### PR DESCRIPTION
**Pull request by asapach**
First step towards progress in #489

### CODE REVIEW

File: test/sourcemaps.js [hunk 1]
```diff
@@ -141,7 +140,7 @@ suite('Source Maps', function() {
           '../fixtures/test-tree/some.js',
           '../fixtures/test-tree/text.txt',
           '../fixtures/test-tree/amd.js',
-          '../fixtures/test-tree/component.jsx',
+          '../fixtures/test-tree/unknown', // component.jsx
```
Comment (guybedford) [inline test/sourcemaps.js]: Is this that sourcemaps issue you were talking about? I may be able to see if I can look into this too.

**Comment by guybedford** [comment]
This is awesome! I can't express how grateful I am to see work here moving forward.

I've given you collaborator access now, so if you want to transfer https://github.com/asapach/babel-plugin-transform-system-register across into the SystemJS organization we can do that. Either way is fine.

File: test/sourcemaps.js (hunk 1 above)
Comment (asapach) [inline test/sourcemaps.js]: Yes, I think previously the sourcemap contained `component.jsx → unknown` (but traceur for some reason ignored the latter), but now babel ignores the intermediate file.

**Comment by asapach** [comment]
Here's what I get when I try to transfer:
transfer
Am I doing this right? Maybe you need to fork it first and grant me the necessary rights there?

**Comment by guybedford** [comment]
@asapach I think you need to initiate the transfer from within the original repo, then I can accept it here?

**Comment by asapach** [comment]
https://help.github.com/articles/transferring-a-repository-owned-by-your-personal-account/
> Users must have admin or owner permissions within the receiving organization before they can transfer a repository that they individually own.

**Comment by guybedford** [comment]
Ah right, added you now.

**Comment by asapach** [comment]
Done. You should also have the NPM admin rights to the package.

**Comment by guybedford** [comment]
Thanks, I'll just aim to run some tests locally here before merge but the approach seems 100%.

Just let me know if you have any other questions around this work.

**Comment by guybedford** [comment]
I think the change is actually a valid fix - the `unknown` is what is reported by the transpiler source map so that is what should be used.

I've merged this in manually now.

It may also be worth abstracting a single `compiler.js` file which encapsulates the standard Babel transpilation up to the plugin parameter, as we're going to be repeating this code a lot otherwise.

**Pull request merged**